### PR TITLE
removed sleeps from tests.. using native cb function instead (non aysnc)

### PR DESCRIPTION
**Pull request by athackst**

### CODE REVIEW

**Review by tinawright — APPROVED** [review]
Love this change!

File: test/test_topic.py [hunk 1]
```diff
@@ -226,29 +219,27 @@ def callback(msg, nodedata):
...
+                                                topic_name="topic_monitor",
                                                 topic_type=String,
                                                 cb=callback,
                                                 blackboard=self.blackboard)
 
         self.blackboard.save("count", 0, self.topic_monitor._id)
-        self.msg_pub = rospy.Publisher("/topic_monitor", String,
-                                       queue_size=1, latch=True)
+
+        assert_equal(rospy.Subscriber, type(self.topic_monitor.topic_sub))
```
Comment (LauraStelzner-TRI) [inline test/test_topic.py:229]: Whats this line checking?  Is it that the type is the same or that they are the same object?

File: test/test_topic.py [hunk 2]
```diff
@@ -402,31 +384,31 @@ def test_msg(self):
 
     def test_msg_cb(self):
         result = self.topic_msg_cb.tick()
-        msg = rospy.wait_for_message('/topic_cb', String)
+        msg = rospy.wait_for_message('topic_cb', String)
```
Comment (LauraStelzner-TRI) [inline test/test_topic.py:387]: whats the reason for the wait_for_message here vs the calling the callback directly?

**Review by LauraStelzner-TRI — COMMENTED** [review]
Looks good, just a few questions before I approve

File: test/test_topic.py (hunk 1 above)
Comment (athackst) [reply test/test_topic.py:229]: it's making sure that self.topic_monitor.topic_sub is a rospy.Subscriber type

File: test/test_topic.py (hunk 2 above)
Comment (athackst) [reply test/test_topic.py:387]: So this test is for the Topic Publisher, so it's making sure an actual topic gets published.  The other one was a Monitor (so it subscribed via a callback).  In that one I'm assuming that ros isn't broken and a ros Subscriber callback is getting appropriately called in the Monitor.  Here, the "topic_cb" is a function that I created, and therefore needs to be tested.

**Review by LauraStelzner-TRI — APPROVED** [review]
Sounds good. Approved.

**Pull request merged**